src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -503,6 +504,32 @@ install_packages_in_root (RpmOstreeTreeComposeContext  *self,
                                                    cancellable, error);
       if (!self->pkgcache_repo)
         return FALSE;
+
+      if (!opt_cachedir)
+        {
+          /* This is part of enabling rpm-ostree inside Docker/Kubernetes/OpenShift;
+           * in this case we probably don't have access to FUSE as today it uses a
+           * suid binary which doesn't have the capabilities it needs.
+           *
+           * So this magical bit tells the core to disable FUSE, which we only do
+           * if --cachedir isn't specified.  Another way to say this is that
+           * running inside an unprivileged container today requires turning off
+           * some of the rpm-ostree intelligence around caching.
+           *
+           * We don't make this actually conditional somehow on running in a
+           * container since if you're not using a persistent cache there's no
+           * real advantage to taking the overhead of FUSE. If the hardlinks are
+           * corrupted, it doesn't matter as they're going to be deleted
+           * anyways.
+           */
+          g_autoptr(GKeyFile) config = ostree_repo_copy_config (self->pkgcache_repo);
+          g_key_file_set_boolean (config, "rpmostree", "enable-fuse", FALSE);
+          if (!ostree_repo_write_config (self->pkgcache_repo, config, error))
+            return FALSE;
+        }
+      else
+        self->unified_core_and_fuse = TRUE;
+
       rpmostree_context_set_repos (self->corectx, self->repo, self->pkgcache_repo);
       self->devino_cache = ostree_repo_devino_cache_new ();
       rpmostree_context_set_devino_cache (self->corectx, self->devino_cache);
```

Comment:
In that case, we should only set up a devino cache if we do have access to FUSE, right? Otherwise, we risk committing the wrong checksum into the tree.